main.c

```diff
@@ -432,6 +432,43 @@ init_headless(struct vkcube *vc)
                  NULL,
                  &b->image);
 
+   VkFormatProperties2 formatProperties = {
+      .sType = VK_STRUCTURE_TYPE_FORMAT_PROPERTIES_2,
+   };
+
+   vkGetPhysicalDeviceFormatProperties2(vc->physical_device,
+		                        vc->image_format,
+					&formatProperties);
+
+   VkFormatFeatureFlags requiredFeatures = 0;
+   requiredFeatures = formatProperties.formatProperties.linearTilingFeatures;
+
+   VkFormatFeatureFlags useFlags = 0;
+   useFlags |= VK_FORMAT_FEATURE_COLOR_ATTACHMENT_BIT;
+
+   if((useFlags & requiredFeatures) != useFlags)
+       fail("Requested image usage is not supported for this format\n");
+
+   VkPhysicalDeviceImageFormatInfo2 formatInfo = {
+      .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2,
+      .format = vc->image_format,
+      .type = VK_IMAGE_TYPE_2D,
+      .tiling = VK_IMAGE_TILING_LINEAR,
+      .usage = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT,
+      .flags = vc->protected ? VK_IMAGE_CREATE_PROTECTED_BIT : 0,
+   };
+
+   VkImageFormatProperties2 imageFormatProperties = {
+      .sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2,
+   };
+
+   VkResult result = vkGetPhysicalDeviceImageFormatProperties2(vc->physical_device,
+		                                               &formatInfo,
+							       &imageFormatProperties);
+
+   if (result != VK_SUCCESS)
+      fail("Image format not supported\n");
```

Comment:
`vkGetPhysicalDeviceFormatProperties2` and `vkGetPhysicalDeviceImageFormatProperties2` must be called before `vkCreateImage`. Their purpose is to verify that the `vkCreateImage` will not trigger undefined behavior.

Also, the code must check that `VkImageCreateInfo::width,height` does not exceed `VkImageFormatProperties::maxWidth,maxHeight`.